src/com/amazon/ion/impl/bin/IonRawBinaryWriter.java

```diff
@@ -125,14 +123,9 @@
 
     private static final byte VARINT_NEG_ZERO   = (byte) 0xC0;
 
-    // See IonRawBinaryWriter#writeString(String) for usage information.
-    static final int SMALL_STRING_SIZE = 4 * 1024;
-
-    // Reusable resources for encoding Strings as UTF-8 bytes
-    final CharsetEncoder utf8Encoder = Charset.forName("UTF-8").newEncoder();
-    final ByteBuffer utf8EncodingBuffer = ByteBuffer.allocate((int) (SMALL_STRING_SIZE * utf8Encoder.maxBytesPerChar()));
-    final char[] charArray = new char[SMALL_STRING_SIZE];
-    final CharBuffer reusableCharBuffer = CharBuffer.wrap(charArray);
+    final Utf8StringEncoder utf8StringEncoder = Utf8StringEncoderPool
+            .getInstance()
+            .getOrCreateUtf8Encoder();
```

Comment:
Rather than allocating several new arrays for each binary writer we construct, we simply pull a `Utf8StringEncoder` from the pool.